articles/150_message_field_names.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: Message Field Name Conflicts
+permalink: articles/message_field_conflicts.html
+abstract:
+  As we move to support more platforms there are we are running into predefined fields and constants that conflict with languages or implementation keywords and macros.
+  This defines how to deal with existing and potential future conflicts.
+published: true
+author: '[Tully Foote](https://github.com/tfoote)'
+categories: Interfaces
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Scope
+
+This article specifies how we will deal with conflicting fields and constants for different languages.
+In the use of the IDL (TODO (tfoote) reference) there will inevitably be conflicting symbols, keywords, or definitions.
+
+## Background and Motivation
+
+This has come up specifically as we're adding support for Windows for ROS2.
+`winnt.h` defines several macros that conflict with existing enumerations. 
+It is also expected to happen as support for new languages are added.
+We cannot know all the potential future keywords and restrictions from a language which is selected to add support for in the future so we must have a generic solution which will allow future languages to be added without disrupting the existing usages.
+
+## New constants
+
+To avoid conflicts generally all constants should be prefixed with a `k`. This will avoid almost all conflicts.
+
+Google Style Guide: https://google.github.io/styleguide/cppguide.html#Constant_Names
+
+TODO(tfoote) Reference ROS2 style guide
+
+## Language Generator Requirements
+
+In the case that a new language is added and there is a detected existing symbol, the language specific generator is expected to implement a basic deconfliction mangling of the symbol to avoid the conflict.
+
+Each generator will define the mangling procedure and provide a list of symbols for which the procedure will be applied.
+This list should be all known keywords or otherwise conflicting symbols.
+
+### Example Deconfliction Policy
+
+A example of a simple name mangling policy would be to append a trailing underscore for conflicting symbols.
+So that would mean that `FOO` if declared as a conflicting symbol would be used in generated code as `FOO_`.
+
+A slightly more unique string is recommended for clarity.
+
+### Stability
+Each language generator will provide a standard name deconfliction policy so that users can use the symbols in a stable manner.
+If a change is needed the generated code should provide a full cycle of backwards compatibility to support changes.
```

Comment:
I don't think this needs to be part of the article. Also in case we discover a collision not known before I don't think there needs / can't be a tick-tock cycle.

Reply:
Removed